test/e2e/deploy_test.go

```diff
@@ -0,0 +1,126 @@
+package e2e_test
```

Comment:
[nit] can we please rename the `deploy_test.go` to something more meaningful, maybe `basic_test.go` or whatever?

Reply:
`configuration_test.go` is ok? it will be contain at least 4